### DISCUSSION TITLE
Diffs for Docx

### DISCUSSION THREAD

**Discussion by avimar** [community/community#27893] [Other Feature Feedback, Questions, & Ideas]
VS code is able to show some of a diff on docx files. 

It seems to work by extracting the raw text and doing a diff on that. It misses formatting changes and headers, but it at least works for much of it.

Can we get that integrated into the PRs and diffs inside github?

There's several tutorials on how to do this for your local git install:
https://stackoverflow.com/questions/22439517/view-docx-file-on-github-and-use-git-diff-on-docx-file-format
https://github.com/vigente/gerardus/wiki/Integrate-git-diffs-with-word-docx-files

**Comment by jfullstackdev** [comment]
> Can we get that integrated into the PRs and diffs inside github?

why would you be needing such ? is the Markdown not enough ?

**Reply by avimar** [reply]
Git hub currently shows:
> BIN -563 KB (4.4%)  > path/File.docx
>Binary file not shown.

I'm not seeing markdown diffs of my docx files.

**Reply by jfullstackdev** [reply]
what do you mean by that ? you are writing Markdown inside .docx file ?

**Reply by avimar** [reply]
I don't know why you're talking about markdown at all.

I have a docx file, and github shows me zero diffs.

It just shows `Binary file not shown.`

**Comment by davidstaheli** [comment]
Hi @avimar! I'm from the GitHub repositories team. Thanks for asking about this.

1. Could you please describe the scenario that leads you to store .docx files in a repository and diff them?
2. Also, could you please describe the type of content in these files? I'm curious if .docx files are giving you benefits that you wouldn't get from other formats.

Thanks!

**Reply by gustavoloureiro** [reply]
Hi @davidstaheli !
I don't know about the author's needs, but in my case, I´m using github to store office documents, which seems to me the most easy and controlable way of doing change management.

**Reply by moabdali** [reply]
Yup, I agree with Gustavo.  I basically want to be able to see my changes between different iterations of my docx files.  I am working on a game where I am updating the rules daily (after doing balance checks or adding new mechanics), and I'd like to see when I made a change or how I'm progressing along.

**Reply by nlisker** [reply]
Similar requirement to the above. Most design, project management, and product management people are not familiar with markdown. They collaborate with the familiar MS Word that has become a standard. Product documentation is uploaded to the repo and it's useful to be able to see who updated what.

**Reply by aritztg** [reply]
We are using docx to store some base templates to generate reports later. Pandoc/markitdown weren't enough to keep some styling. That is why we store those base templates in the repository. Having a nice diff would be helpful.

For those that might be interested, there is this way to perform this with ease in local git: https://github.com/marcocrowe/track-ms-word-docx-with-github
